packages/form/core/form-group.ts

```diff
@@ -18,4 +18,14 @@ export class FormGroup {
 	get(name: string): FormControl | undefined {
 		return this.controls.find((control) => control.name === name);
 	}
+
+	setValue(values: object) {
+		this.controls.map((control) => {
+			for (const value in values) {
+				if (value == control.name) {
+					control.setValue(values[value]);
+				}
+			}
+		});
+	}
```

Comment:
You got the gist of the solution! However, we can still improve this, performance-wise.

Usually, you want to avoid having a loop within a loop for performance. 😉

Try to do this by just looping through the names in your values object and using the FormGroup.get function to get the appropriate control.

Something like:

```typescript
Object.keys(values).forEach(valueName => this.get(valueName)?.setValue(values[valueName]));
```

Reply:
Changed the solution. Thanks for the great suggestion.